crates/oxc_linter/src/rules/eslint/func_names.rs

```diff
@@ -0,0 +1,507 @@
+use oxc_ast::ast::{
+    AssignmentTarget, AssignmentTargetProperty, BindingPatternKind, Expression, Function,
+    FunctionType, PropertyKind,
+};
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Default, Clone)]
+pub struct FuncNames {
+    default_config: FuncNamesConfig,
+    generators_config: FuncNamesConfig,
+}
+
+#[derive(Debug, Default, Clone, PartialEq)]
+enum FuncNamesConfig {
+    #[default]
+    Always,
+    AsNeeded,
+    Never,
+}
+
+impl FuncNamesConfig {
+    pub fn from(raw: &str) -> Self {
+        match raw {
+            "always" => FuncNamesConfig::Always,
+            "as-needed" => FuncNamesConfig::AsNeeded,
+            "never" => FuncNamesConfig::Never,
+            _ => FuncNamesConfig::default(),
```

Comment:
We should tell consumers when their configs are incorrect instead of failing silently.

Reply:
I agree they're doing this. This is a failing of existing configuration parsing. IMO, we've avoided panics in many cases where the user should be notified of misconfiguration.

An example, as requested: https://github.com/oxc-project/oxc/blob/6ff200d072fe55cb720c8dbe3a00f885fd1bbd28/crates/oxc_linter/src/rules/eslint/no_unused_vars/options.rs#L329

Reply:
I don't really understand what the expecting result will be.
Here it what is does now:

>failures:
> ---- rules::eslint::func_names::test stdout ----
> thread 'rules::eslint::func_names::test' panicked at crates/oxc_linter/src/rules/eslint/func_names.rs:376:71:
called `Result::unwrap()` on an `Err` value: OxcDiagnostic { inner: OxcDiagnosticInner { message: "Expecting always, as-needed or never, got as-needed2", labels: None, help: None, severity: Error, code: OxcCode { scope: None, number: None } } }